tools/test/examples/examples_lib.py

```diff
@@ -444,3 +476,18 @@ def update_mbedos_version(config, tag, examples):
     
     return 0
   
+def save_test_spec(json_spec, name="test_spec.json"):
+    """save the given json data to test_spec.json"""
+    print ("Dumping json test_specs file {}".format(name))
+    with open(name, 'w') as outfile:  
+        json.dump(json_spec, outfile , indent=4)
+        
+def fetch_output_image(output):
+    """find the mbed build image from thet last 5 lines of a given log """
```

Comment:
nit pick on spelling:
```suggestion
    """Find the build image from the last 5 lines of a given log"""
```